lib/order.rb

```diff
@@ -0,0 +1,81 @@
+require 'csv' #ruby reads the csv file
+require_relative 'customer' #knows about customer class
+
+#assume there is only one of each product
+class Order
+
+  attr_reader :id
+  attr_accessor :products,  :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id #number
+    @products = products #{"melon" => 9.99, "chips" => 3.00} ] cost from cvs?
+    @customer = customer #from customer class
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError.new("Invalid fulfillment status.") if [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status) != true
+  end
+
+  def total
+    if @products.empty?
+      return 0
+    else
+      total = @products.values.sum
+      total = (total*1.075).round(2)
+    end
+    return total
+  end
```

Comment:
Consideration for refactoring......

You have a conditional that checks if the `@products` hash is empty. If it is, then `return 0`. This makes sense. Otherwise, you'll have to do `@products.values.sum` in line 23.

However, what does `.values` return if `@products` is an empty hash?
`.values` on an empty hash actually gives back an empty array! (Check this in `irb`: Run `{}.values` and you'll get back `[]`).

What does `.sum` on an empty array give back? `[].sum` actually gives back.... `0`!!!!!!!
If `total` is `0`, and we go to line 24... `total = (0*1.075).round(2)` will actually give back..... `0.0`!

It turns out, even if we DON'T check for `@products.empty?`, your `total` logic will still calculate a total cost of 0 correctly. AKA this code functions the same as your code above:

```suggestion
  def total
    total = @products.values.sum
    total = (total*1.075).round(2)
    return total
  end
```